packages/server/app/Controllers/Http/MyProfileController.ts

```diff
@@ -1,21 +1,17 @@
 import { HttpContextContract } from '@ioc:Adonis/Core/HttpContext';
-import {
-    GetMyProfileInformationRequestBody,
-    GetMyProfileInformationResponseBody,
-} from '@musicroom/types';
-import User from 'App/Models/User';
+import { GetMyProfileInformationResponseBody } from '@musicroom/types';
 import invariant from 'tiny-invariant';
 
 export default class MyProfileController {
     public async getMyProfileInformation({
-        request,
+        auth,
     }: HttpContextContract): Promise<GetMyProfileInformationResponseBody> {
-        const rawBody = request.body();
+        const user = auth.user;
+        invariant(
+            user !== undefined,
+            'User must be authenticated to get her profile information',
```

Comment:
**🤓 nitpick:** I'm not sure if the error message is accurate. Unless I'm mistaken if the user is not authenticated then adonis would throw an error before ?
I have no clue on how should be this error but at the first read it looks like the authentication assertion is made manually here and not by the adonis auth middleware. Let me know what you think

Reply:
I think that the method of the controller has no idea whether an authentication middleware has been called before it.
At least TypeScript does not, and `auth.user` can be `undefined`.
This is why I named the error that way, because there is no strict link contract between the method and the route, which requires at its level the auth middleware.